### PR TITLE
fix(rs): resume sessions by explicit agent_session_id

### PR DESCRIPTION
**Pull request by maui1911**
## Summary

Fixes the session-resume path so each session reattaches to the SPECIFIC conversation instead of the agent's most-recent fallback. Mirrors C# `SessionManager.CreateAgentSession` resume branch + `JoinResumeByIdArgs`.

User report: reopening sessions in CodeScope rs was issuing `pi -c` instead of `pi --session <id>`, with equivalent regressions across Claude / Copilot / OpenCode.

## Changes

- New pure helper `codescope_core::build_resume_auto_type(profile, agent_session_id)` selects:
  - `[command, resume_by_id_args[0..n-1].., resume_by_id_args[last] + id_suffix]` when both id and `resume_by_id_args` are present. Last token concat'd without a space when it ends with `=` (Copilot's `--resume=<id>`), space-separated otherwise (Claude `--resume <id>`, OpenCode `--session <id>`, Pi `--session <id>`).
  - `[command, resume_args...]` otherwise (Pi `-c`, Copilot `--continue`, bare `claude`, Codex `resume`).
- `AppShell::reopen_session` (sidebar history reopen) now passes `restored.agent_session_id` through the helper.
- `LayoutState::RestoreTab` carries the persisted session id; the cold-start rehydrate path in `rehydrate_or_cold_start` looks up the session row in `projects.json` and re-derives the resume auto-type from `Session.agent_id + agent_session_id`. Falls back to the legacy `auto_type` (and a fresh-spawn path) when the layout entry has no session id or the row was removed.
- Cross-group tab drag (`move_tab_to_group`) reparents the existing pty in-process — no auto-type re-evaluation needed.

## Tests

- 12 new unit tests in `codescope_core::agent_registry` covering every built-in agent (Claude / Codex / OpenCode / Copilot / Pi) in both with-id and without-id shapes, plus edge cases (empty id treated as no-id, empty `command` returns `None`, multi-token `resume_by_id_args` appends id to the last token).
- New layout test for legacy `RestoreTab` rows without a `session_id` field.
- `cargo test --workspace` clean (571 tests pass).
- `cargo build --workspace` clean.

## Follow-up (out of scope)

The discovery layer (claude/copilot/opencode/pi) currently updates `Tab::adopted_session_id` in memory but does NOT persist it back to `Session.agent_session_id` in `projects.json`. As a result, resume-by-id only fires today for sessions whose id was stamped through the C# build's `agentSessionId` flow (which the Rust port hasn't ported yet). Filing as a separate issue per the user's instructions.

## Test plan

- [ ] Open Claude session, close tab, reopen from history → `claude --resume <id>` is typed at the prompt.
- [ ] Open Pi session, close tab, reopen from history → `pi --session <id>` (not `pi -c`).
- [ ] Open Copilot session, close tab, reopen from history → `copilot --resume=<id>` (no space).
- [ ] Open OpenCode session, close tab, reopen from history → `opencode-cli --session <id>`.
- [ ] Close + reopen the app with a tab strip → tabs rehydrate using `resume_by_id_args + id` when the persisted session row carries `agent_session_id`.
- [ ] Sessions without a stored `agent_session_id` still rehydrate via `resume_args` (existing behaviour) — no regression for legacy projects.json rows.

Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>

### CODE REVIEW

File: codescope-rs/src/app.rs [hunk 1]
```diff
@@ -1759,6 +1759,14 @@ impl AppShell {
...
                     group_index: g_idx,
                     active_in_group: t_idx == group.active_tab,
+                    // Persist the session id so the rehydrate path on
+                    // next launch can look up `agent_session_id` and
+                    // resume the *specific* conversation, not just
+                    // "most recent". Matches the C# build's behaviour
+                    // where layout-rehydrate goes through
+                    // `CreateAgentSession(resume: true, agentSessionId:
+                    // stored.AgentSessionId)`.
+                    session_id: Some(tab.session_id.clone()),
```
[the remark this answers is not in the document]
Comment (maui1911) [reply codescope-rs/src/app.rs]: Fixed in 95ca8d3 — `snapshot_open_tabs` now only writes `session_id = Some(...)` when `lookup_session_by_id` resolves the row in the current `self.projects` snapshot. Free-floating tabs (no project context at spawn) persist with `session_id = None` so the rehydrate path falls back to a fresh-spawn instead of adopting a ghost id.

File: codescope-rs/src/app.rs [hunk 2]
```diff
@@ -1796,19 +1804,48 @@ impl AppShell {
...
+            // Re-bind to the stored session row when the layout
+            // entry carries an id and the row still exists in
+            // `projects.json`. Otherwise we fall back to a fresh
+            // session id (`spawn_tab_in` mints one via
+            // `allocate_session_id`) — the legacy behaviour for
+            // pre-resume-by-id layouts.
+            let restore_session_id =
+                stored_session.as_ref().map(|s| s.id.clone()).or_else(
+                    || tab.session_id.clone(),
+                );
```
[the remark this answers is not in the document]
Comment (maui1911) [reply codescope-rs/src/app.rs]: Fixed in 95ca8d3 — `restore_session_id` is now strictly `stored_session.as_ref().map(|s| s.id.clone())`, no fallback to the raw layout id. When the row is gone (project removed, race with sidebar write) we pass `None` and `allocate_session_id` mints + appends a fresh row via `SessionManager::open`.
